src/server/cluster/cluster_shard_migration.cc

```diff
@@ -57,16 +66,84 @@ void ClusterShardMigration::FullSyncShardFb(Context* cntx) {
   DCHECK(leftover_buf_);
   io::PrefixSource ps{leftover_buf_->InputBuffer(), Sock()};
 
-  uint8_t ok_buf[4];
-  ps.ReadAtLeast(io::MutableBytes{ok_buf, 4}, 4);
+  JournalReader reader{&ps, 0};
+  TransactionReader tx_reader{};
 
-  if (string_view(reinterpret_cast<char*>(ok_buf), 4) != "SYNC") {
-    VLOG(1) << "FullSyncShardFb incorrect data transfer";
-    cntx->ReportError(std::make_error_code(errc::protocol_error),
-                      "Incorrect FullSync data, only for tets");
+  while (!cntx->IsCancelled()) {
+    if (cntx->IsCancelled())
+      break;
+
+    auto tx_data = tx_reader.NextTxData(&reader, cntx);
+    if (!tx_data)
+      break;
+
+    TouchIoTime();
+
+    if (!tx_data->is_ping) {
+      ExecuteTxWithNoShardSync(std::move(*tx_data), cntx);
+    } else {
+      // TODO check about ping logic
+    }
+  }
+}
+
+void ClusterShardMigration::ExecuteTxWithNoShardSync(TransactionData&& tx_data, Context* cntx) {
+  if (cntx->IsCancelled()) {
+    return;
   }
 
-  VLOG(1) << "FullSyncShardFb finished after reading 4 bytes";
+  bool inserted_by_me = tx_data.IsGlobalCmd() &&
+                        multi_shard_exe_->InsertTxToSharedMap(tx_data.txid, tx_data.shard_cnt);
+
+  if (tx_data.shard_cnt <= 1 || !tx_data.IsGlobalCmd()) {
```

Comment:
What is the reason to create a sync between different shards on executing in cluster migration?